### PR TITLE
Add pod_ip_metrics collector to netd

### PR DESCRIPTION
**Pull request by sdmodi**
This new module emits metrics for pod IP allocation by examining the gke-pod-network directory

### CODE REVIEW

File: pkg/metrics/collector/pod_ip_metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	defer f.Close()
+	if _, err := f.WriteString(content); err != nil {
+		t.Fatalf("Error writing to file %s: %v", path, err)
+	}
+}
+
+func mustCreateIpAddrDir(t *testing.T, dir string) string {
+	tempDir, err := ioutil.TempDir("", dir)
+	if err != nil {
+		t.Fatalf("error creating dir %s: %v", dir, err)
```
Comment (MrHohn) [inline pkg/metrics/collector/pod_ip_metrics_test.go:24]: nit: error -> Error

File: pkg/metrics/collector/pod_ip_metrics.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	c.usedIpv6AddrCount = ipv6Count
+	c.dualStackCount = dualCount
+	c.dualStackErrorCount = dualErrCount
+	c.duplicateIpCount = dupIpCount
+	return nil
+}
+
+func (c *podIpMetricsCollector) Update(ch chan<- prometheus.Metric) error {
+	if err := c.listIpAddresses(gkePodNetworkDir); err != nil {
+		return nil
```
Comment (MrHohn) [inline pkg/metrics/collector/pod_ip_metrics.go:165]: Humm, probably log the error here instead? I know listIpAddresses() already logs that - but at the first glance it seems like omitting the error without looking into that func.

File: pkg/metrics/collector/pod_ip_metrics.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+	s := bufio.NewScanner(buf)
+	s.Scan()
+	return s.Text(), s.Err()
+}
+
+func (c *podIpMetricsCollector) listIpAddresses(dir string) error {
+	f, err := os.Open(dir)
+	if err != nil {
+		glog.Errorf("Error opening directory %s, %v", dir, err)
```
Comment (MrHohn) [inline pkg/metrics/collector/pod_ip_metrics.go:106]: Just wanted to point out there may be a short period of time when the `gke-pod-network` folder doesn't exist yet (e.g. no pod has been created or only hostnetwork pods are created). That however should be fine as we only log an error and retry later.

**Review by MrHohn — APPROVED** [review]
Thanks Sudeep, seems straight forward. Just a few nits - this LGTM.

File: pkg/metrics/collector/pod_ip_metrics.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type ipFamily int
+
+const (
+	gkePodNetworkDir = "/host/var/lib/cni/networks/gke-pod-network"
+	ipv4 ipFamily = 0
```
[the remark this answers is not in the document]
Comment (sdmodi) [reply pkg/metrics/collector/pod_ip_metrics.go:34]: Not sure what you mean. I did add a String function for this type.

File: pkg/metrics/collector/pod_ip_metrics.go (hunk 3 above)
Comment (sdmodi) [reply pkg/metrics/collector/pod_ip_metrics.go:106]: Yes. I noticed that and this is ok.

File: pkg/metrics/collector/pod_ip_metrics.go (hunk 2 above)
Comment (sdmodi) [reply pkg/metrics/collector/pod_ip_metrics.go:165]: Done

**Review by MrHohn — APPROVED** [review]
/lgtm
/approve

File: pkg/metrics/collector/pod_ip_metrics.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (MrHohn) [reply pkg/metrics/collector/pod_ip_metrics.go]: Yep changing to use iota is what I meant. Thx.

**Pull request merged**